SYCL/Config/select_device.cpp

```diff
@@ -1,46 +1,42 @@
 // RUN: %clangxx -fsycl %s -o %t.out
 //
-// RUN: env WRITE_DEVICE_INFO=1 %GPU_RUN_PLACEHOLDER %t.out
-// RUN: env READ_DEVICE_INFO=1 %GPU_RUN_PLACEHOLDER %t.out
+// RUN: env WRITE_DEVICE_INFO=1 SYCL_DEVICE_FILTER=%u %t.out
+// RUN: env READ_DEVICE_INFO=1 SYCL_DEVICE_FILTER=%u %t.out
```

Comment:
It is necessary to use "%GPU_RUN_PLACEHOLDER" because this value is going to be replaced by proper SYCL_DEVICE_FILTER automatically by LIT. "SYCL_DEVICE_FILTER=%u" results in an error (per testing logs) because %u value doesn't exist. Could you please revert this part of changes.

Removing "XFAIL: cuda || hip" below is enough to enable this test for CUDA and HIP.